databuilder/tests/unit/models/test_application.py

```diff
@@ -93,21 +95,35 @@ def setUp(self) -> None:
             ),
         )
 
-        # Test several different potential application types
-        for application_type in ['Databricks', 'Snowflake', 'EMR']:
+        # Test several non-airflow applications
+        AppTestCase = namedtuple('AppTestCase', ['name', 'generates_table'])
+        non_airflow_cases = [
+            AppTestCase(name='Databricks', generates_table=False),
+            AppTestCase(name='Snowflake', generates_table=True),
+            AppTestCase(name='EMR', generates_table=False),
+        ]
+
+        for case in non_airflow_cases:
+            application_type = case.name
             url = f'https://{application_type.lower()}.com/job/1234'
             id = f'{application_type}.hive.test_table'
             description = f'{application_type} application for hive.test_table'
-
-            # task_id and dag_id are irrelevant for non Airflow applcations for for backwards
-            # compatibility we do not want to remove the init arguements
-            application = Application(
-                task_id='',
-                dag_id='',
+            table_key = TableMetadata.TABLE_KEY_FORMAT.format(
+                db='hive',
+                cluster='gold',
                 schema='default',
-                table_name='test_table',
-                application_url_template=url,
+                tbl='test_table',
+            )
+
+            application = GenericApplication(
```

Comment:
@jroof88 This shows what change you'd have to make on your end.
The key difference is that the key you introduced in https://github.com/amundsen-io/amundsen/pull/1398 `application://{application_type}/{database}/{table}` is replaced by an ability to specify a key override. 

The new generic key has been changed to `application://{application_type}/{application_id}`. Because the key is independent of the table, it lets the generic model support many<>many relationships out of the box.